core/src/main/java/com/aem/demo/core/services/impl/RssFeedServiceImpl.java

```diff
@@ -41,4 +51,87 @@ public List<NewsCard> readFeed() {
         return feed.getEntries().stream().map(NewsCard::new)
                 .collect(Collectors.toList());
     }
+
+    @Override
+    public String saveRssFeedNodes(List<NewsCard> cardList) {
+        String nodeCreated= StringUtils.EMPTY;
+        try {
+            ResourceResolver resolver = ResolverUtil.newResolver(resourceResolverFactory);
+            Session session = resolver.adaptTo(Session.class);
+
+            if (session.nodeExists(NODE_PATH)){
+                nodeCreated = addNewsCardNode(session, cardList);
+            } else {
+                addParentNode(session);
+                nodeCreated = addNewsCardNode(session, cardList);
+            }
+
+        } catch (Exception e) {
+            log.error("\n Error while creating node - {} ",e.getMessage());
+        }
+        return nodeCreated;
+    }
+
+    private int count = 0;
+
+    private String addNewsCardNode(Session session, List<NewsCard> list){
+
+        try {
+            for (NewsCard elm:list) {
+                count++;
+                String nodeName= count + " newscard_node";
+                Node node = session.getNode(NODE_PATH);
+                Node cardNode = node.addNode(nodeName, NEWSCARDNODE_TYPE);
+                cardNode.setProperty("topic", elm.getTopic());
+                cardNode.setProperty("article", elm.getArticle());
+                cardNode.setProperty("link", elm.getLink());
+                cardNode.setProperty("pubDate", String.valueOf(elm.getPubDate()));
+                cardNode.setProperty("image", elm.getImage());
+                saveImage(cardNode, elm.getImage());
+                session.save();
+            }
+
+
+        } catch (Exception e) {
+            log.error("\n Error while creating NewsCard node ");
+        }
+        return null;
+    }
+
+    private String addParentNode(Session session){
+        try {
+            if(session.nodeExists("/content/aemtraining/us/en")){
```

Comment:
This must be a constant.
Moreover, this constant is much related to the `NODE_PATH = "/content/aemtraining/us/en/us-newscard"` (almost the same path)

Reply:
And above all, you are going to have language copies of the cards. Us/en copy is going to be just one of the copies. Because us/en is just one of the possible languages) That's why it's better to download content to `language-masters/en`, not to `us/en`